File: Polimorfismo/Ejercicio5_java/java/Celular.java
package Polimorfismo.Ejercicio5.python;

public class Celular {
    private int numeroDeTelefono;
    private String dueño;
    private double espacio;
    private double ram;
    private int numeroApp;

    public Celular(int numeroDeTelefono, String dueño, double espacio, double ram, int numeroApp) {
        this.numeroDeTelefono = numeroDeTelefono;
        this.dueño = dueño;
        this.espacio = espacio;
        this.ram = ram;
        this.numeroApp = numeroApp;
    }

    public void iadd() {
        this.numeroApp += 10;
    }

    public void dsub() {
        this.espacio -= 5;
        this.espacio = Math.round(this.espacio * 100.0) / 100.0; // rounding to 2 decimal places
    }

    @Override
    public String toString() {
        return "nro: " + this.numeroDeTelefono + " - dueño: " + this.dueño + " - espacio: " + this.espacio + " GB - ram: " + this.ram + " GB - nroApps: " + this.numeroApp;
    }

    public static void main(String[] args) {
        Celular celular = new Celular(78929330, "Joel", 6.44, 10, 20);
        System.out.println(celular.toString());
        celular.iadd();
        celular.dsub();
        System.out.println(celular.toString());
    }
}

File: Polimorfismo/Ejercicio5_java/Celular.py
"""
    Ejercicio 5. Dada la clase celular <nroTel, dueño, espacio, ram, nroApp>
a) realizar el diagrama de clases
b) realizar la sobrecarga del operador ++ para aumentar en 10 el nroApp.
c) realizar la sobrecarga del operador - - para disminuir el espacio en 5 gb.
d) mostrar los datos antes y después de los operadores.
"""

class Celular:
    def __init__(self, nroDeTelefono, nombre, espacio, ram, nroApp):
        self.__nroTelefono = nroDeTelefono
        self.__dueño = nombre
        self.__espacio = espacio
        self.__ram = ram
        self.__nroApp = nroApp
    
    def iadd(self):
        self.__nroApp += 10
    
    def dsub(self):
        self.__espacio -= 5
        self.__espacio = round(self.__espacio, 2)
    
    def __str__(self):
        return f"nro: {self.__nroTelefono} - dueño: {self.__dueño} - espacio: {self.__espacio} GB - ram: {self.__ram} GB - nroApps: {self.__nroApp}"
    

celular = Celular(78929330, "Joel", 6.44, 10, 20)
print(celular.__str__())
celular.iadd()
celular.dsub()
print(celular.__str__())